src/compas_timber/_fabrication/step_joint.py

```diff
@@ -160,9 +160,20 @@ def heel_depth(self, heel_depth):
     def step_shape(self):
         return self._step_shape
 
-    @step_shape.setter  # TODO: should this be defined automatically depending on the other parameters? (ie. if heel_depth > 0 and step_depth > 0, then step_shape = "double")
+    @step_shape.setter
     def step_shape(self, step_shape):
-        if step_shape not in [StepShapeType.DOUBLE, StepShapeType.STEP, StepShapeType.HEEL, StepShapeType.TAPERED_HEEL]:
+        if step_shape == StepShapeType.DOUBLE:
+            if self.step_depth <= 0 or self.heel_depth <= 0:
+                raise ValueError("For a 'double' step_shape, both step_depth and heel_depth must be greater than 0.")
+        elif step_shape == StepShapeType.STEP:
+            if self.step_depth <= 0 or self.heel_depth != 0:
+                raise ValueError("For a 'step' step_shape, step_depth must be greater than 0 and heel_depth must be 0.")
+        elif step_shape in [StepShapeType.HEEL, StepShapeType.TAPERED_HEEL]:
+            if self.heel_depth <= 0 or self.step_depth != 0:
+                raise ValueError(
+                    "For 'heel' or 'tapered heel' step_shape, heel_depth must be greater than 0 and step_depth must be 0."
+                )
+        else:
```

Comment:
I would actually not put this here.
validating it here means that, depending on the chronological order the parameters are set you might be  validating parameters that haven't been set yet. 

As by the end of `__init__` all parameters should have already properly been set. that would be a good place to call a function which validates all these.